### ISSUE TITLE
Analytics Implementation or QA Support for [Design System - New Components]

### ISSUE THREAD

**Issue by k80bowman** [department-of-veterans-affairs/va.gov-team#45510]
# Google Analytics Implementation or QA Request

## What this form is for

<details>
<summary>More Information</summary>
<div>
    <br>
<p>Use this template to request Google Analytics implementation or QA for your product. Please fill out all brackets.</p>
    
> _Please see the following information for implementation timeline:_
> New Teams: Implementation will usually be completed in 4 sprints
> 
> - Sprint 1: Discovery/Orientation
> - Sprint 2: FE Implementation
> - Sprint 3: Tagging Implementation
> - Sprint 4: QA
> 
> All other teams: Implementation will usually be completed in 2 sprints
> 
> - Sprint 1: Discovery and FE Implementation
> - Sprint 2: Tagging Implementation and QA

</div>
</details>

### Required
- **Is this request for **Google Analytics implementation** or **QA**?**

     - [x] Google Analytics implementation
     - [x] QA

- **Who is this request for?**

    - Design System Team (@department-of-veterans-affairs/platform-design-system-team)
    
    
- **Provide the link(s) for Google Analytics implementation or QA**
  
    - [Insert link(s)]
    
- **When is the launch date?**
    - The components are already available for use

- **What interactions need to be tracked?** _NOTE: [Design System Components](https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/Design-System-Components.1840808040.html) are automatically tracked_
    
    - [Insert other interactions - please review the [Design System Components](https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/Design-System-Components.1840808040.html) and [GA Data Dictionary](https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/Google-Analytics-Data-Dictionary.1810464912.html)  to make sure these interactions aren't already tracked]
        - Date
            - Fires when the date input loses focus after its value was changed
            - Sensitive component, analytics are not enabled by default
        - Banner
            - Fires when an anchor link inside the banner or the banner dismiss icon is clicked
        - Button
            - Fires when the button is clicked
        - Button Pair
            - Fires when the primary button is clicked
            - Fires when the secondary button is clicked
        - Link
            - Fires when the link is clicked
        - Memorable Date
            - Fires when the memorable date loses focus after its value was changed 
            - Sensitive component, analytics are not enabled by default
        - Number Input
            - Fires when the component loses focus
            - Sensitive component, analytics are not enabled by default
        - Pagination
            - Fires when the previous button is clicked, fires when the next button is clicked, and fires when a page is selected
            - Sensitive component, analytics are not enabled by default
        - Promo Banner
            - Fires when an anchor link inside the banner or the banner dismiss icon is clicked

### Optional
- **If applicable, what test users are needed?**  _NOTE: Do not put staging credentials in this ticket; store or reference them in a .md file in the `va.gov-team-sensitive` repository_

    - [Insert test user]

- **Any additional comments**

    - [Insert comments]

> Please leave the following blank

## Acceptance criteria
- [ ] All appropriate issue tagging is completed
- [ ] QA has been completed
- [ ] Events have been added to `Product` and `Benefit Hub` Content Groups

## Additional information

Please refer to [Platform Collaboration Cycle](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/index.html) or the [Analytics Request Touchpoint](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Analytics-request.1782120453.html) on Platform website for more information about the Collaboration Cycle.

**Comment by jonathan-epstein13** [comment]
Hello @k80bowman! Can you provide the URL for the specific tracking? Thank you!

**Comment by k80bowman** [comment]
These are design system components, they get used all over the site, I don't have one. Or maybe I'm not understanding exactly what you need.

**Comment by jonathan-epstein13** [comment]
Hello @k80bowman just following up on the ticket. 
Here is the [data dictionary](https://depo-platform-documentation.scrollhelp.site/analytics-monitoring/google-analytics-data-dictionary) for the events for implementation.
The existing events for clicks and buttons can be used.
Implement the existing navigation tracking that you and your team feel suits your needs best. 

Let me know if you have any other questions and once that is complete, if you can provide specific pages login user or a way to test out the events for you and your team.

**Comment by michelle-dooley** [comment]
@k80bowman - I'm going to schedule some time for us to go over this request so we can make we are both aligned on the ask and what we need to provide.  cc @jonathan-epstein13

**Comment by k80bowman** [comment]
Thank you @michelle-dooley!

**Comment by k80bowman** [comment]
@jonathan-epstein13 from looking at the data dictionary you posted, I was able to use the tracking information for the button, button pair, pagination, and promo banner components. So, the components I still don't have tracking information for are:
- Date
- Banner
- Link
- Memorable date
- Number input

**Comment by k80bowman** [comment]
Could I add the new telephone component to this list as well please? It fires when the telephone number is clicked.

**Comment by jonathan-epstein13** [comment]
@k80bowman just want to make sure about the new telephone component for the list isn't being tracked and needs implementation? However, it was firing when the number is clicked? Thank you for clarifying.

**Comment by k80bowman** [comment]
I don't see a data layer specification for it, so it is not currently being tracked. Our component is implemented such that the analytics event will fire when the number is clicked. Does that help?

**Comment by jonathan-epstein13** [comment]
Yes, that helps. If I have any other questions, I'll be sure to reach out @k80bowman. Thank you!

**Comment by k80bowman** [comment]
All of the components should be in the Storybook, that is where we display our components and provide examples for engineers. We don't have automatic tracking set up in Storybook right now.

**Comment by michelle-dooley** [comment]
@k80bowman - Here are the specs for Banner (va-banner).  It should work/track the same as the Promo banner (va-promo-banner).  

Component Name: ‘va-banner’:
Action:  ‘linkClick’,
Event:  ‘nav-banner-link-click’,
Prefix:  ‘banner’.
 
 
Data Layer Specification:
'event': 'nav-banner-link-click'
'banner-click-label': //populate according to link click text
'banner-click-target': //populate according to target in component
'banner-click-href': //populate according to click href
'banner-closeable': //populate with true or false based on if promo banner is capable of being closed
'banner-type': //populate according to banner type (icon) in component`
 
Event Action:
Navigation - Banner - {{DL - banner-type}} - {{DL - banner-href}} - {{DL - romo-banner-target}} - {{DL - banner-text}}

**Comment by michelle-dooley** [comment]
@k80bowman - Here are specs for Link (va-link).  Let me know once you get set up and I will test.

Component Name:  ‘va-link’
Action:  ‘click’
Event:  ^nav-link-click’
Prefix:  ‘link’
link-click-label:  //populate according to link click text

**Comment by k80bowman** [comment]
@michelle-dooley I wanted to let you know that I will be passing this ticket off to @jamigibbs. I had a question about the banner specs above, though. Could I get specs for a 'nav-banner-close' event please? Thank you!

**Comment by jamigibbs** [comment]
@michelle-dooley Just a quick follow-up on the components that we still need GA tracking for here. I've added a few more details for each one but let me know if there's anything else's needed for them. 

- [ ]  **Date (va-date)**

Example: https://design.va.gov/storybook/?path=/docs/components-va-date--default
- Fires when the date input loses focus after its value was changed.
- Sensitive component, analytics are not enabled by default.
- Component info:
```
    componentName: 'va-date',
    action: 'blur',
    details: {
        year: this.year,
        month: this.month,
        day: this.day,
        monthYearOnly: this.monthYearOnly, // Whether or not only the Month and Year inputs should be displayed.
    },
```

- [ ]   **Banner (va-banner)**

Example: https://design.va.gov/storybook/?path=/docs/components-va-banner--default#dismissible
- We already have a click event for this. We just need one for when the banner "X" is clicked to close it. For example, 'nav-banner-close'


- [ ]   **Button Pair (va-button-pair)**

Example: https://design.va.gov/storybook/?path=/docs/components-va-button-pair--default
- Fires when the primary button is clicked
- Fires when the secondary button is clicked
- Component info:
```
    componentName: 'va-button-pair',
    action: 'click',
    details: {
        primaryLabel: this.primaryLabel,
        secondaryLabel: this.secondaryLabel
    },
```

- [ ]   **Memorable Date (va-memorable-date)**

Example: https://design.va.gov/storybook/?path=/docs/components-va-memorable-date--default
- Fires when the memorable date loses focus after its value was changed
- Sensitive component, analytics are not enabled by default
Component info:

```
    componentName: 'va-memorable-date',
    action: 'blur',
    details: {
        label: this.primaryLabel,
        yearNum: this.secondaryLabel
        monthNum: this.monthNum
        dayNum: this.dayNum
    },
```

- [ ]   **Number Input (va-memorable-date)**

Example: https://design.va.gov/storybook/?path=/docs/components-va-number-input--default
- Fires when the component loses focus
- Sensitive component, analytics are not enabled by default
Component info:

```
    componentName: 'va-number-input',
    action: 'blur'
    details: {
        label: this.label,
        value: this.value,
    }
```

- [ ]   **Promo Banner (va-promo-banner)**

Example: https://design.va.gov/storybook/?path=/docs/components-va-promo-banner--default
- Fires, when an anchor link inside the banner or the banner dismiss icon, is clicked
- We will need both a click event for the link and a click event for the close
- For example, 'nav-promo-banner-click' and 'nav-promo-banner-close'

Component info:
```
    componentName: 'va-promo-banner',
    action: 'click'
    details: {
        text: this.el.innerText,
        href: this.href,
        type: this.type, //  'announcement', 'news', 'email-signup'
    }
```

**Comment by michelle-dooley** [comment]
@jamigibbs - I think this is everything you need for the new components.  I also updated the GA4 worksheet with these new events for that implementation.

va-date:  event: 'int-date-blur', prefix: 'date'
va-banner:  event:  'int-banner-close', prefix: 'banner'
va-button-pair:  event: 'int-button-pair-click', prefix: 'button-pair'
va-memorable-date:  event: 'int-memorable-date-blur', prefix:  'memorable-date'
va-number-input:  event: 'int-number-input-blur', prefix: 'number-input'
va-promo-banner:  event: 'int-promo-banner-close', prefix: 'promo-banner'

**Comment by michelle-dooley** [comment]
@jamigibbs - just checking in on this one.  Do you know when you will be ready for me to test?

**Comment by jamigibbs** [comment]
@michelle-dooley We have a concern about components "double logging" that's holding this up a bit. Do you have any insight on how that would appear on the GA dashboard? We're assuming that the `va-button` and `va-button-pair` events would both fire at the same time and that wouldn't be a good thing for tracking: https://github.com/department-of-veterans-affairs/component-library/pull/513#discussion_r1023206983

**Comment by michelle-dooley** [comment]
@jamigibbs - Yes I agree with the concern of double counting.  I don't know within your code if there is a away to only trigger the button pair action and not the single button click when you know it is part of a pair?!?  That is something we don't have control over.  I am not overly familiar with button pairs, why would you need track that differently than just tracking all button clicks individually?  I assume if one is a subset of another and you see someone triggered that event then you know the parent button that was clicked to get them there.

**Comment by michelle-dooley** [comment]
@jamigibbs - just checking in on this one.  Is there anything ready for me to test yet?

**Comment by jamigibbs** [comment]
Hi @michelle-dooley thanks for checking in! We are still working on getting the [PR](https://github.com/department-of-veterans-affairs/component-library/pull/513) approved but I'm trying to get it expedited.

**Comment by jamigibbs** [comment]
@michelle-dooley These updates have been merged into vets-website with the latest component library release [13.5.0](https://github.com/department-of-veterans-affairs/component-library/releases/tag/v13.5.0). Several of them may not be easily testable though (any components related to forms) in staging or prod because they need an `enable-analytics` flag to be active on the component which I can not find any examples of. I have tested all of these locally though and the GA event is firing based on the adswerve browser extension.

I'm comfortable closing this ticket based on my local testing but if you'd like to try some other testing in staging or prod, let me know if there's anything I can help with.
